python/ray/util/sgd/torch/torch_runner.py

```diff
@@ -130,12 +121,28 @@ def _toggle_profiling(self, profile=False):
             self.timers.disable()
         self.training_operator._set_timers(self.timers)
 
-    def state_dict(self):
+    def state_dict(self, to_cpu=False):
         """Returns the state of the runner."""
+        model_states = []
+        for model in self.models:
+            sd = model.state_dict()
+            if to_cpu:
+                for k, v in sd.items():
+                    sd[k] = v.cpu()
+            model_states.append(sd)
+        optimizer_states = []
+        for opt in self.optimizers:
+            sd = opt.state_dict()
+            if to_cpu:
+                for state in sd["state"].values():
+                    for k, v in state.items():
+                        if torch.is_tensor(v):
+                            state[k] = v.cpu()
+            optimizer_states.append(sd)
         state = {
             "epoch": self.epochs,
             "operator": self.training_operator.state_dict(),
-            "models": [model.state_dict() for model in self.models],
+            "models": model_states,
             "optimizers": [opt.state_dict() for opt in self.optimizers]
```

Comment:
optimier_states

Reply:
yes good catch. missed this when copying over from cluster.